src/Areas/Monitor/Services/MetricsQueryClientService.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Monitor.Query;
+using AzureMcp.Options;
+using AzureMcp.Services.Azure;
+using AzureMcp.Services.Azure.Tenant;
+
+namespace AzureMcp.Areas.Monitor.Services;
+
+public class MetricsQueryClientService(ITenantService tenantService) : BaseAzureService(tenantService), IMetricsQueryClientService
+{
+    public async Task<MetricsQueryClient> CreateClientAsync(string? tenant = null, RetryPolicyOptions? retryPolicy = null)
+    {
+        var credential = await GetCredential(tenant);
+        var options = AddDefaultPolicies(new MetricsQueryClientOptions());
+
+        if (retryPolicy != null)
+        {
+            options.Retry.Delay = TimeSpan.FromSeconds(retryPolicy.DelaySeconds);
+            options.Retry.MaxDelay = TimeSpan.FromSeconds(retryPolicy.MaxDelaySeconds);
+            options.Retry.MaxRetries = retryPolicy.MaxRetries;
+            options.Retry.Mode = retryPolicy.Mode;
+            options.Retry.NetworkTimeout = TimeSpan.FromSeconds(retryPolicy.NetworkTimeoutSeconds);
+        }
+
+        return new MetricsQueryClient(credential, options);
```

Comment:
MetricsQueryClient uses the legacy ARM endpoint. There's a new data-plane endpoint for querying metrics. Please use the new client. See sample here - https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.Query#query-metrics-for-multiple-resources

Reply:
This is a very significant change and it also causes some hard dependencies on the endpoint name. I would like to do this in a separate PR, is that OK?